olp-cpp-sdk-core/src/client/OlpClient.cpp

```diff
@@ -286,5 +305,163 @@ CancellationToken OlpClient::CallApi(
       [cancel_context]() { cancel_context->CancelOperation(); });
 }
 
+std::pair<http::NetworkResponse, std::shared_ptr<std::stringstream>>
+SendRequest(const http::NetworkRequest& request,
+            const olp::client::OlpClientSettings& settings,
+            const olp::client::RetrySettings& retry_settings,
+            client::CancellationContext context) {
+  http::NetworkResponse network_response = kCancelledErrorResponse;
+  auto interest_flag = std::make_shared<std::atomic_bool>(true);
+
+  Condition condition{};
+
+  auto network = settings.network_request_handler;
+  auto response_body = std::make_shared<std::stringstream>();
+
+  http::SendOutcome outcome =
+      network->Send(request, response_body,
+                    [&, interest_flag](http::NetworkResponse response) {
+                      if (interest_flag->exchange(false)) {
+                        network_response = std::move(response);
+                        condition.Notify();
+                      }
+                    });
+
+  if (outcome.IsSuccessful()) {
```

Comment:
What happens with the negative outcome? Some of the cases are very severe.
Usually on negative outcome you should have something like:

```cpp
if (!send_outcome.IsSuccessful()) {
    callback(HttpResponse(static_cast<int>(send_outcome.GetErrorCode()),
                          ErrorCodeToString(send_outcome.GetErrorCode())));
    return CancellationToken();
  }
```

Reply:
Negative branch on 355 line:
```c++
 network_response =
        http::NetworkResponse()
            .WithStatus(static_cast<int>(outcome.GetErrorCode()))
            .WithError(ErrorCodeToString(outcome.GetErrorCode()));
```

Reply:
Ok, didn't see that.